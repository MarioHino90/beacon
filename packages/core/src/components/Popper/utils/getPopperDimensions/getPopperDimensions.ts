import { normalizeDimension } from "../normalizeDimension";

/**
 *
 * @param popperRef
 */
export const getPopperDimensions = (
  popperRef: React.RefObject<HTMLElement>
): [number, number] => {
  // get the dimensions of the Popper element
  const elementHeight = normalizeDimension(
    popperRef.current?.getBoundingClientRect().height || 0
  );
  const elementWidth = normalizeDimension(
    popperRef.current?.getBoundingClientRect().width || 0
  );

  return [elementHeight, elementWidth];
};
