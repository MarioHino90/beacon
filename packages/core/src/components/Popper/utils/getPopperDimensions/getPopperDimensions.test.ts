import { getPopperDimensions } from "./getPopperDimensions";

describe(`getPopperDimensions`, () => {
  it(`should match snapshot`, () => {
    // create a fake popper
    const popper = document.createElement("div");
    const content = document.createTextNode("popper");

    popper.appendChild(content);

    // make assertion
    expect(getPopperDimensions({ current: popper })).toMatchSnapshot();
  });
});
