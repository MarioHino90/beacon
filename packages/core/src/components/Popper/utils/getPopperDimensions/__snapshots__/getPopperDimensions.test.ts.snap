// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getPopperDimensions should match snapshot 1`] = `
[
  0,
  0,
]
`;
